### PR TITLE
Switch to UBI9 minimal and install required version of JDK

### PR DESCRIPTION
**Pull request by wendigo**
This change makes the docker image build against a fixed, required version of JDK (instead of the latest).

PR also fixes inconsistency between the `air.java.version` property in parent pom, `.java-version` file, and `TrinoSystemRequirements.verifyJavaVersion()`.

In Trino's testing `docker-images` we currently use version `17.0.5`.

### CODE REVIEW

File: .java-version [hunk 1]
```diff
@@ -1 +1 @@
-17.0
+17.0.3
```
Comment (electrum) [inline .java-version]: Let's use `17.0.9` as that is the latest. `17.0.3` is 18 months old.

File: core/trino-main/src/main/java/io/trino/server/TrinoSystemRequirements.java [hunk 2]
```diff
@@ -93,7 +93,7 @@ else if ("Mac OS X".equals(osName)) {
 
     private static void verifyJavaVersion()
     {
-        Version required = Version.parse("17.0.3");
+        Version required = Version.parse("17.0.4");
```
Comment (electrum) [inline core/trino-main/src/main/java/io/trino/server/TrinoSystemRequirements.java:96]: The commit message doesn't match this change. `air.java.version` is unchanged, and there are actually three separate versions here, which we should be careful about not conflating:

* minimum required version for users to run Trino
* minimum required version for building Trino
* the version used for Docker images

Plus whatever else might use `.java-version` (I manage my Java versions manually so I'm not familiar with exactly how the version managers work)

I would be fine with making the last two the same, as it seems reasonable for developers to build and test against the version that we ship in the Docker image. But we need to be careful about bumping the required Java version, as that can impact users who manually install Java and might not need the latest version. We typically only bump it if there are crash bugs or other egregious issues. Though I'm open to discussing changing this policy to be more aggressive if it avoids support issues than it causes.

File: core/docker/Dockerfile [hunk 3]
```diff
@@ -13,16 +13,31 @@
...
+ARG JDK_DOWNLOAD_LINK
+ARG JDK_VERSION
+ENV JAVA_HOME="/usr/lib/jvm/jdk-${JDK_VERSION}"
+
+RUN \
+    set -xeuo pipefail && \
+    microdnf install -y tar gzip && \
+    # Install JDK from the provided archive link \
+    echo "Downloading JDK from ${JDK_DOWNLOAD_LINK}" && \
+    curl -#LfSo /tmp/jdk.tar.gz "${JDK_DOWNLOAD_LINK}" && \
```
Comment (electrum) [inline core/docker/Dockerfile:26]: Why not pipe the output of `curl` to `tar`?

File: core/docker/Dockerfile [hunk 4]
```diff
@@ -13,16 +13,31 @@
...
 
-# Use Eclipse Temurin as they have base Docker images for more architectures.
-FROM eclipse-temurin:17-jdk
+FROM registry.access.redhat.com/ubi9/ubi-minimal:latest AS jdk-download
+ARG JDK_DOWNLOAD_LINK
+ARG JDK_VERSION
+ENV JAVA_HOME="/usr/lib/jvm/jdk-${JDK_VERSION}"
+
+RUN \
+    set -xeuo pipefail && \
```
Comment (electrum) [inline core/docker/Dockerfile:22]: No need for `pipefail` if we don't have a pipe (but maybe we should have one)

File: core/trino-main/src/main/java/io/trino/server/TrinoSystemRequirements.java (hunk 2 above)
Comment (martint) [reply core/trino-main/src/main/java/io/trino/server/TrinoSystemRequirements.java:96]: Just to clarify, when you say "minimum required version for building Trino", you're referring to the version of the JDK used to compile Trino, not the version of the JDK the code relies on (i.e., language level, Java APIs, JVM features), right?

Comment (wendigo) [reply core/trino-main/src/main/java/io/trino/server/TrinoSystemRequirements.java:96]: @martint we have 4 different source of the JDK version:

1) `air.java.version` - maven check whether minimum JDK version is used to compile code
2) `TrinoSystemRequirements` - runtime check whether minimum JDK verion is used to compile code
3) `project.build.targetJdk` - language level (`--release` passed to the compiler)
4) `.java-version` - JDK version to used for local development/docker image

**Comment by wendigo** [comment]
PTAL @martint @electrum

**Pull request closed without merging**